lnbits/wallets/strike.py

```diff
@@ -0,0 +1,467 @@
+                return InvoiceResponse(False, None, None, "Invalid invoice response")
+
+            self.pending_invoices.append(invoice_id)  # Add invoice ID to pending invoices.
+            return InvoiceResponse(True, invoice_id, bolt11, None)  # Return successful invoice response.
+        except httpx.HTTPStatusError as e:
+            msg = e.response.json().get("message", e.response.text)  # Get error message from response.
+            return InvoiceResponse(False, None, None, f"Strike API error: {msg}")
+        except Exception:
+            logger.exception("Error in create_invoice()")
+            return InvoiceResponse(False, None, None, "Connection error")
+
+    async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
+        try:
+            idem = str(uuid.uuid4())
+            # 1) Create a payment quote.
+            q = await self._post(
+                "/payment-quotes/lightning",
+                json={"lnInvoice": bolt11},
+                headers={**self.client.headers, "idempotency-key": idem},
+            )
+            quote_id = q.json().get("paymentQuoteId")  # Get the payment quote ID.
+            if not quote_id:  # Check if the quote ID is present.
+                return PaymentResponse(
+                    False, None, None, None, "Strike: missing paymentQuoteId"
+                )
+
+            # 2) Execute the payment quote.
+            e = await self._patch(f"/payment-quotes/{quote_id}/execute")
+            data = e.json() if e.content else {}  # Parse JSON response or use an empty dictionary.
+            payment_id = data.get("paymentId")  # Get the payment ID.
+            state = data.get("state", "").upper()  # Get the payment state and convert it to uppercase.
+
+            # Network fee → msat.
+            fee_obj = data.get("lightningNetworkFee") or data.get("totalFee") or {}  # Get fee object.
+            fee_btc = Decimal(fee_obj.get("amount", "0"))  # Get fee amount in BTC.
+            fee_msat = int(fee_btc * Decimal(1e11))  # Convert fee from BTC to millisatoshis.
+
+            # Store mapping for later polling.
+            if payment_id:  # If payment ID is present.
+                self.pending_payments[payment_id] = quote_id
+
+            if state in {"SUCCEEDED", "COMPLETED"}:  # If payment succeeded.
+                preimage = data.get("preimage") or data.get("preImage")  # Get payment preimage.
+                return PaymentResponse(True, payment_id, fee_msat, preimage, None)  # Return successful payment response.
+
+            # Strike often returns 202/PENDING immediately; treat as “still working”.
+            if state in {"PENDING", "QUEUED", "READY_TO_SETTLE", ""}:  # If payment is still pending.
+                return PaymentResponse(None, payment_id, None, None, None)  # Return pending payment response.
+
+            return PaymentResponse(False, payment_id, None, None, f"State: {state}")  # Return failed payment response with state.
```

Comment:
Isn't there an explicit list of states for 'FAILED'?
If a new `pending` status is introduced by the API then it will not be recognized and the payment will be marked as failed (which has an impact on the wallet balance).

It is better to check explicitly for `failed` and leave the default branch for `pending`.